src/TimePicker/Clock.js

```diff
@@ -20,10 +20,16 @@ class Clock extends Component {
   };
 
   state = {
-    selectedTime: this.props.initialTime || new Date(),
+    selectedTime: null,
     mode: 'hour',
   };
 
+  componentWillMount() {
+    this.setState({
+      selectedTime: this.props.initialTime,
```

Comment:
True but `||` and `defaultProps` don't have the same behavior. Try out the TimePicker example is you don't trust me :grin:.